config/envoyconfig/luascripts/clean-upstream.lua

```diff
@@ -1,15 +1,22 @@
-function remove_pomerium_cookie(cookie_name, cookie)
-    -- lua doesn't support optional capture groups
-    -- so we replace twice to handle pomerium=xyz at the end of the string
-    cookie = cookie:gsub(cookie_name .. "=[^;]+; ", "")
-    cookie = cookie:gsub(cookie_name .. "=[^;]+", "")
-    return cookie
-end
-
 function has_prefix(str, prefix)
     return str ~= nil and str:sub(1, #prefix) == prefix
 end
 
+function remove_pomerium_cookie(cookie_name, cookie)
+    local result = ""
+    for c in cookie:gmatch("([^;]+)") do
+        c = c:gsub("[ ]+", "")
+        if not has_prefix(c, cookie_name) then
```

Comment:
I think this may remove other cookies too (e.g. `_pomerium_something_else` in addition to `_pomerium`).

Should we also split on the first `=`? Would something like this work?

```suggestion
        local name = c:match("^ *([^=]+)")
        if name ~= cookie_name then
```